packages/angular/cli/commands/update-impl.ts

```diff
@@ -178,17 +178,20 @@ export class UpdateCommand extends Command<UpdateCommandSchema> {
         return false;
       }
 
+      this.logger.info(colors.green(`${colors.symbols.check} Migration succeeded.`));
+
       // Commit migration
-      if (commit) {
-        let message = `${packageName} migration - ${migration.name}.`;
-        if (migration.description) {
-          message += '\n' + migration.description;
-        }
-        // TODO: Use result.files once package install tasks are accounted
-        this.createCommit(message, []);
+      const commitPrefix = `${packageName} migration - ${migration.name}.`;
+      const commitMessage = migration.description
+          ? `${commitPrefix}\n${migration.description}`
+          : commitPrefix;
+      // TODO: Use result.files once package install tasks are accounted
+      const hash = commit ? this.createCommit(commitMessage, []) : null;
+      if (hash) {
+        this.logger.info(
```

Comment:
I think it may be a good idea to log if no changes were made as well and remove any ambiguity as to whether a commit happened or not.

Reply:
I added a log when a commit is not made, though this pulled the thread on a lot of complex error handling tasks. Hypothetically, any shell command can fail for arbitrary reasons. Previously the CLI just swallowed any `git` errors and silently continued. This could cause problems if one `git commit` fails, but the next succeeds, then the commit will contain extra files with the wrong message.

While working on this, I came across a few edge cases and did my best to handle/fix them:

1. If a commit fails, the CLI swallowed the error and silently continued.
    * Now the CLI logs the command's `stderr` and aborts.
1. A migration may not change any files, `git commit` would then fail because there is nothing to commit.
    * Now the CLI checks if there are changes to commit and just logs if there aren't any.
1. The command to look up the HEAD commit hash may fail, so we can't print it.
    * This is a non-fatal error, the CLI just prints a slightly less descriptive message and logs a warning that something weird happened, since this is unlikely to occur in practice.
1. I also noticed that one migration contained a commit message with a double quote character and was failing due to Bash expansion.
    * Changed the `git commit` command to read the commit message from `stdin`, where Bash escaping isn't a concern.

I did my best to minimize the nest if-statements, but that was difficult to do here. Let me know if you see any opportunities to simplify this.

I also chose not to log that a commit was skipped due to CLI option. I think that is becoming the default now, and I felt it was confusing to explicitly log that we are following default behavior for users who probably aren't even aware of the flag.